pkg/etcd/etcd.go

```diff
@@ -904,7 +904,16 @@ func (e *ETCD) listSnapshots(ctx context.Context, snapshotDir string) ([]Snapsho
 			return nil, err
 		}
 
-		objects := e.s3.client.ListObjects(ctx, e.config.EtcdS3BucketName, minio.ListObjectsOptions{})
+		s3FolderInUse := e.config.EtcdS3Folder != ""
+
+		var loo minio.ListObjectsOptions
+		if s3FolderInUse {
+			loo = minio.ListObjectsOptions{
+				Recursive: true,
+			}
```

Comment:
Pass in the prefix, there's no point in listing the whole bucket from the root if you know everything's going to be under this folder prefix
```suggestion
			loo = minio.ListObjectsOptions{
				Prefix: e.config.EtcdS3Folder,
				Recursive: true,
			}
```

Reply:
Sadly, none of the prefix stuff is working in this library or is in a very non-obvious way. The prefix was the preferred route.

Reply:
Hmm, listing with a prefix is pretty important to the S3 API. Do you perhaps need to strip the trailing slash or something? Client-side filtering by checking the key is going to be terrible; buckets can contain millions of objects.

Reply:
https://docs.min.io/docs/golang-client-api-reference#:~:text=prefix%3A%20%22myprefix%22

Reply:
if the object is bucket/foo/bar.txt, the prefix should be specified as "foo", not "/foo" or "foo/".

Reply:
Was the missing recursive field. ugh